cmd/abc/abc.go

```diff
@@ -104,9 +107,56 @@ func setLogEnvVars() {
 	}
 }
 
+// checkForUpdates asynchronously checks for updates and prints results to
+// returned buffer.
+// A sync.WaitGroup is returned so main can wait for completion before exiting.
+func checkForUpdates(ctx context.Context) func() {
+	updatesCh := make(chan string, 1)
+
+	go func() {
```

Comment:
Can abc-updater hide this complexity so that callers don't have to each implement this kind of concurrency?

It's really hard to get the concurrency details right (channels, goroutines, and selects), so better to just implement it once and have everyone call it.

I realize this contradicts the standard Go API advice of letting callers handle concurrency and keeping the library simple. But in this case I think it's worth it, because most people struggle to implement correct concurrent code of this type, and this library is for our internal use where we can dictate a consistent style.

I'm imagining something like:

```
func main() {
  report := abcupdater.Check(appID, version.Version)
  defer report()
}
```

Reply:
Assuming `report` is a closure with the channel that will do the select, and `Check` would set up the channel and kick off the goroutine?

Reply:
Yeah, that's what I had in mind.

Reply:
Also, the `abcupdater.Check()` function could be a wrapper around a simpler non-concurrent API like `func OtherCheck(appID, version string) (string, error)`. The caller could choose to deal with concurrency or not.